api/tests/unit/domain/usecases/get-progression_test.js

```diff
@@ -175,11 +175,14 @@ describe('Unit | Domain | Use Cases | get-progression', () => {
         assessmentId,
         userId,
       });
+      const competenceSkills = [domainBuilder.buildSkill()];
 
       beforeEach(() => {
         sandbox.stub(assessmentRepository, 'getByAssessmentIdAndUserId').resolves(competenceEvaluationAssessment);
         sandbox.stub(competenceEvaluationRepository, 'getByAssessmentId').resolves(competenceEvaluation);
-        sandbox.stub(skillRepository, 'findByCompetenceId').resolves([]);
+        sandbox.stub(skillRepository, 'findByCompetenceId').resolves(competenceSkills);
```

Comment:
rien à voir avec la PR, mais je me demande pourquoi on utilise `sandbox.stub` et non `sinon.stub`.

Reply:
Intéressant!
- D'après la doc `Sandboxes removes the need to keep track of every fake created, which greatly simplifies cleanup.` (https://sinonjs.org/releases/v2.2.0/sandbox/).
- Je pense que `sandbox` est utile lorsque l'on veut avoir un stub de service générique, mais overridé pour certaines cas de test. Par exemple ici, on a défini un certain comportement pour `improvementService` dans un `beforeEach`
```
sandbox.stub(improvementService, 'filterKnowledgeElementsIfImproving')
          .withArgs({ knowledgeElements: [], assessment: competenceEvaluationAssessment }).returns([]);
```
Mais dans un cas de test spécifique, on a overridé son comportement via:
```
improvementService.filterKnowledgeElementsIfImproving
            .withArgs({ knowledgeElements, assessment: competenceEvaluationAssessment }).returns(knowledgeElementsFiltered);
```

Reply:
@celineung ce que tu décris est possible sans `sandbox`, on peut toujours redéfinir le comportement d'un _stub_ après sa création.

Comme l'indique la doc de Sinon.JS les _sandboxes_ sont très pratique pour éviter d'avoir à "débrancher" (`stub.restore()`)  les _stubs_ un par un à la fin du test. C'est pourquoi on veut en utiliser à peu près tout le temps dans nos tests.

Mais depuis la version 5 de Sinon.JS l'objet global `sinon` se comporte lui-même comme une _sandbox_, ce qui enlève en plus la nécessité de la créer au début du test (`sinon.createSandbox()`) et permet de réinitialiser tous les _stubs_ créés pendant le test avec un simple `sinon.restore()` dans un `afterEach`. De plus, il est possible de définir un `afterEach` global qui appelle ce `sinon.restore()` ce qui fait que les tests peuvent créer des _stubs_ à volonté sans avoir à se préoccuper de leur réinitialisation qui aura lieu systématiquement. C'est ce que nous avons mis en place (en enlevant en même temps tous les appels à `createSandbox`) dans https://github.com/1024pix/pix/pull/314 ; depuis, il n'y a plus de raison d'appeler `createSandbox` dans les tests, sauf cas très particulier comme [celui-ci](https://github.com/1024pix/pix/blob/e575364bb5a3ed93020c879e31b5ea2d3cf724d5/api/tests/acceptance/application/saml-controller_test.js#L163).

Malheureusement il semble que depuis l'intégration de https://github.com/1024pix/pix/pull/314, de nouveaux appels à `createSandbox` aient été introduits, probablement en suivant d'anciennes habitudes.

En bref, pour répondre à ta question @HEYGUL il n'y a pas de bonne raison d'utiliser `sandbox` au lieu de `sinon` dans ce test, il pourrait être simplifié en utilisant directement `sinon` et en supprimant la déclaration de `sandbox` et le `afterEach` correspondant. Et de même dans la poignée d'autres fichiers de test qui utilisent encore `createSandbox`.